ooni/nettests/blocking/web_connectivity.py

```diff
@@ -0,0 +1,546 @@
+            result['status']['failure'] = failureToString(failure)
+            self.report['tcp_connect'].append(result)
+
+        return d
+
+    @defer.inlineCallbacks
+    def control_request(self, sockets):
+        self.control = yield self.web_connectivity_client.control(
+            http_request=self.input,
+            tcp_connect=sockets
+        )
+        self.report['control'] = self.control
+
+    @defer.inlineCallbacks
+    def experiment_http_get_request(self):
+        log.msg("* doing HTTP(s) request {}".format(self.input))
+        retries = 0
+        while True:
+            try:
+                result = yield self.doRequest(self.input,
+                                              headers=REQUEST_HEADERS)
+                break
+            except:
+                if retries > self.localOptions['retries']:
+                    log.debug("Finished all the allowed retries")
+                    raise
+                log.debug("Re-running HTTP request")
+                retries += 1
+
+        defer.returnValue(result)
+
+    def compare_headers(self, experiment_http_response):
+        control_headers_lower = {k.lower(): v for k, v in
+                self.report['control']['http_request']['headers'].items()
+        }
+        experiment_headers_lower = {k.lower(): v for k, v in
+            experiment_http_response.headers.getAllRawHeaders()
+        }
+
+        uncommon_ctrl_headers = (set(control_headers_lower.keys()) -
+                                 set(COMMON_SERVER_HEADERS))
+        uncommon_exp_headers = (set(experiment_headers_lower.keys()) -
+                                set(COMMON_SERVER_HEADERS))
+
+        return len(uncommon_ctrl_headers.intersection(
+                            uncommon_exp_headers)) > 0
+
+    def compare_body_lengths(self, experiment_http_response):
+        control_body_length = self.control['http_request']['body_length']
+        experiment_body_length = len(experiment_http_response.body)
+
+        if control_body_length == experiment_body_length:
+            rel = float(1)
+        elif control_body_length == 0 or experiment_body_length == 0:
+            rel = float(0)
+        else:
+            rel = float(control_body_length) / float(experiment_body_length)
+
+        if rel > 1:
+            rel = 1/rel
+
+        self.report['body_proportion'] = rel
+        if rel > float(self.factor):
+            return True
+        else:
+            return False
+
+    def compare_titles(self, experiment_http_response):
+        experiment_title = extractTitle(experiment_http_response.body).strip()
+        control_title = self.control['http_request']['title'].strip()
+        first_exp_word = experiment_title.split(' ')[0]
+        first_ctrl_word = control_title.split(' ')[0]
+        if len(first_exp_word) < 5:
+            # We don't consider to match words that are shorter than 5
+            # characters (5 is the average word length for english)
+            return False
```

Comment:
Perhaps, if the first word is shorter than five you could check the second, and so on?